packages/phishing-controller/src/PhishingController.ts

```diff
@@ -566,6 +574,60 @@ export class PhishingController extends BaseController<
     }
   }
 
+  /**
+   * Scan a URL for phishing. It will only scan the hostname of the URL. It also only supports
+   * web URLs.
+   *
+   * @param url - The URL to scan.
+   * @returns The phishing detection scan result.
+   */
+  async scanUrl(url: string): Promise<PhishingDetectionScanResult> {
+    if (
+      !url.toLowerCase().startsWith('http://') &&
+      !url.toLowerCase().startsWith('https://')
+    ) {
+      throw new Error('invalid web url');
+    }
+
+    const hostname = getHostnameFromUrl(url);
+    if (!hostname) {
+      throw new Error('invalid web url');
+    }
+
+    const response = await safelyExecuteWithTimeout(
+      async () => {
+        const res = await fetch(
+          `${PHISHING_DETECTION_BASE_URL}/${PHISHING_DETECTION_SCAN_ENDPOINT}?url=${encodeURIComponent(hostname)}`,
+          {
+            method: 'GET',
+            headers: {
+              Accept: 'application/json',
+            },
+          },
+        );
+        if (!res.ok) {
+          // Instead of throwing inside safelyExecuteWithTimeout, return an error object
+          return {
+            error: `${res.status} ${res.statusText}`,
+          };
+        }
+        const data = await res.json();
+        return data;
+      },
+      true,
+      8000,
+    );
+
+    // Need to do it this way because safelyExecuteWithTimeout returns undefined for both timeouts and
+    if (!response) {
+      throw new Error('request failed: Request timed out');
+    } else if ('error' in response) {
+      throw new Error(`request failed: ${response.error}`);
+    }
+
+    return response;
```

Comment:
We probably should not throw and buble the error. Handling the error in the clients seems like a bigger effort and not aligned with how the current PhishingController.test is being used.